File: src/main/java/com/natu/ftax/ledger/infrastructure/LedgerEntryRepositoryJdbc.java
package com.natu.ftax.ledger.infrastructure;

import com.natu.ftax.ledger.domain.LedgerEntry;
import org.springframework.stereotype.Repository;

@Repository
public class LedgerEntryRepositoryJdbc implements LedgerEntryRepository {

    private final LedgerEntryRepositoryJpa ledgerEntryRepositoryJpa;

    public LedgerEntryRepositoryJdbc(LedgerEntryRepositoryJpa ledgerEntryRepositoryJpa) {
        this.ledgerEntryRepositoryJpa = ledgerEntryRepositoryJpa;
    }

    @Override
    public void save(LedgerEntry ledgerEntry, String ledgerBookId) {
        ledgerEntryRepositoryJpa.save(LedgerEntryEntity.fromDomain(ledgerEntry, ledgerBookId));
    }
}

File: src/main/kotlin/com/natu/ftax/ledger/infrastructure/LedgerEntryRepositoryJdbc.kt
package com.natu.ftax.ledger.infrastructure

import com.natu.ftax.ledger.domain.LedgerEntry
import org.springframework.stereotype.Repository

@Repository
class LedgerEntryRepositoryJdbc(val ledgerEntryRepositoryJpa: LedgerEntryRepositoryJpa) : LedgerEntryRepository {
    override fun save(ledgerEntry: LedgerEntry, ledgerBookId: String) {
        ledgerEntryRepositoryJpa.save(LedgerEntryEntity.fromDomain(ledgerEntry, ledgerBookId))
    }
}